RadixWallet/Prelude/Extensions/Collection+Extra.swift

```diff
@@ -32,3 +32,19 @@ extension OrderedSet where Element: Hashable {
 		}
 	}
 }
+
+extension MutableCollection {
+	/// Mutates in place the elements of the collection
+	public mutating func mutateAll(_ mutate: (inout Self.Element) -> Void) {
+		for index in indices {
+			mutate(&self[index])
+		}
+	}
+}
+
+extension RangeReplaceableCollection {
+	/// Filters in place the elements of the collection
```

Comment:
This still creates a new array though, right? I suppose you could loop over the array and remove the elements you don't want?